### PR TITLE
Bump packages and replace jayschema

### PR DESCRIPTION
**Pull request by Half-Shot**
This is awaiting me doing a release for matrix-appservice-node, but anyway this resolves a lot of our security issues.

### CODE REVIEW

File: lib/components/config-validator.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 "use strict";
-var fs = require("fs");
-var yaml = require("js-yaml");
-var JaySchema = require("jayschema");
-var extend = require("extend");
+const fs = require("fs");
+const yaml = require("js-yaml");
+const validator = require('is-my-json-valid');
```
Comment (turt2live) [inline lib/components/config-validator.js]: quotes are different

Comment (turt2live) [reply lib/components/config-validator.js]: also can we `UpperCamelCase` the validator given it's a class?

Comment (turt2live) [reply lib/components/config-validator.js]: but it's a class that gives a function... gah why is js so horrible sometimes :/ I still think upper camel case is the way to go here, even if the package is weird.

File: package.json [hunk 2]
```diff
@@ -26,19 +26,20 @@
   "dependencies": {
     "bluebird": "^2.9.34",
     "extend": "^3.0.0",
-    "jayschema": "^0.3.1",
-    "js-yaml": "^3.4.0",
-    "matrix-appservice": "^0.3.4",
+    "is-my-json-valid": "^2.19.0",
+    "js-yaml": "^3.12.0",
+    "matrix-appservice": "matrix-org/matrix-appservice-node#73839c6077b92b4cd0d52ec8dee599a96142d3b3",
```
Comment (turt2live) [inline package.json]: I'd rather us wait for the package to be released before we merge this PR

File: lib/components/config-validator.js (hunk 1 above)
Comment (Half-Shot) [reply lib/components/config-validator.js]: I was making it conform to the others, but yea

File: package.json (hunk 2 above)
Comment (Half-Shot) [reply package.json]: Seems fair

**Pull request merged**